projects/go-lib/src/lib/components/go-search/go-search.component.ts

```diff
@@ -0,0 +1,82 @@
+import { Component, ElementRef, HostListener, OnInit } from '@angular/core';
+import { AnimationEvent } from '@angular/animations';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+
+import { searchLoaderAnim, searchResultsAnim } from '../../animations/search.animation';
+import { GoSearchService } from './go-search.service';
+
+@Component({
+  selector: 'go-search',
+  templateUrl: './go-search.component.html',
+  styleUrls: ['./go-search.component.scss'],
+  animations: [searchLoaderAnim, searchResultsAnim]
+})
+export class GoSearchComponent implements OnInit {
+
+  goSearchForm: FormGroup;
+  searchActive: boolean = false;
+  resultsOverflow: string = 'hidden';
+
+  @HostListener('document:click') onDocumentClick(event) {
```

Comment:
HostListener doesn't appear to pass any arguments by default. You can configure and array of arguments to be passed to the `onDocumentClick` function as the second argument.

We can use this to pass the `$event.target` rather than the whole `$event` object, which is a [bad practice in Angular](https://angular.io/guide/user-input#passing-event-is-a-dubious-practice).

Consider this:
```javascript
@HostListener('document:click', ['$event.target'])
private onDocumentClick(target: HTMLElement): void {
  this.closeSearchEvent(target);
}
```

Reply:
@AlexOverbeck interesting, I understand what they're getting at here. This is actually broke, lol. I implemented this approach and fixed the issue. Good catch.